x/posts/internal/keeper/querier.go

```
@@ -90,3 +100,31 @@ func queryPosts(ctx sdk.Context, req abci.RequestQuery, keeper Keeper) ([]byte,
 
 	return bz, nil
 }
+
+//queryPollAnswers handles the request to get poll answers related to a post with given id
+func queryPollAnswers(ctx sdk.Context, path []string, _ abci.RequestQuery, keeper Keeper) ([]byte, error) {
+	id, err := types.ParsePostID(path[0])
+
+	if err != nil {
+		return nil, sdkerrors.Wrap(sdkerrors.ErrUnknownRequest, fmt.Sprintf("Invalid post id: %s", path[0]))
+	}
+
+	_, found := keeper.GetPost(ctx, id)
+	if !found {
+		return nil, sdkerrors.Wrap(sdkerrors.ErrUnknownRequest, fmt.Sprintf("Post with id %s not found", id))
+	}
+
+	pollAnswers := keeper.GetPostPollAnswersDetails(ctx, id)
+
+	pollAnswersResponse := types.PollAnswersQueryResponse{
```

Should we return an error if the post does not have an associated poll?